### PR TITLE
Upgrade python to 3.10.0

### PR DESCRIPTION
**Pull request by nvuillam**
Upgrade python to 3.10.0

### CODE REVIEW

**Comment by Kurt-von-Laven** [comment]
Black recently ([literally 14 hours ago](https://github.com/psf/black/issues/2242#issuecomment-974832792)) added support for Python 3.10 at version 21.11b1, and it looks like your automation already picked up that change, so I believe this PR is unblocked unless there is another linter that can't yet handle Python 3.10 that I'm unaware of.

**Comment by nvuillam** [comment]
@Kurt-von-Laven good catch, let's see what CI says :)

**Comment by nvuillam** [comment]
## [MegaLinter](https://megalinter.github.io) status: :warning: [**WARNING**](https://github.com/megalinter/megalinter/actions/runs/1488844712)

|   Descriptor   |                                                  Linter                                                   |Files|Fixed|                                  Errors                                  |Elapsed time|
|----------------|-----------------------------------------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------------------|------------|
|✅ BASH         |[bash-exec](https://megalinter.github.io/beta/descriptors/bash_bash_exec)                                  |    6|     |                                                                         0|1.23s       |
|✅ BASH         |[shellcheck](https://megalinter.github.io/beta/descriptors/bash_shellcheck)                                |    6|     |                                                                         0|1.45s       |
|✅ BASH         |[shfmt](https://megalinter.github.io/beta/descriptors/bash_shfmt)                                          |    6|    0|                                                                         0|1.25s       |
|✅ COPYPASTE    |[jscpd](https://megalinter.github.io/beta/descriptors/copypaste_jscpd)                                     |yes  |     |no                                                                        |2.43s       |
|✅ CREDENTIALS  |[secretlint](https://megalinter.github.io/beta/descriptors/credentials_secretlint)                         |yes  |     |no                                                                        |3.14s       |
|✅ DOCKERFILE   |[dockerfilelint](https://megalinter.github.io/beta/descriptors/dockerfile_dockerfilelint)                  |    1|     |                                                                         0|0.43s       |
|✅ DOCKERFILE   |[hadolint](https://megalinter.github.io/beta/descriptors/dockerfile_hadolint)                              |    1|     |                                                                         0|0.99s       |
|✅ GIT          |[git_diff](https://megalinter.github.io/beta/descriptors/git_git_diff)                                     |yes  |     |no                                                                        |0.2s        |
|✅ JSON         |[eslint-plugin-jsonc](https://megalinter.github.io/beta/descriptors/json_eslint_plugin_jsonc)              |   20|    0|                                                                         0|2.36s       |
|✅ JSON         |[jsonlint](https://megalinter.github.io/beta/descriptors/json_jsonlint)                                    |   19|     |                                                                         0|9.22s       |
|✅ JSON         |[v8r](https://megalinter.github.io/beta/descriptors/json_v8r)                                              |   20|     |                                                                         0|17.5s       |
|✅ MARKDOWN     |[markdownlint](https://megalinter.github.io/beta/descriptors/markdown_markdownlint)                        |  196|    0|                                                                         0|5.18s       |
|✅ MARKDOWN     |[markdown-link-check](https://megalinter.github.io/beta/descriptors/markdown_markdown_link_check)          |  196|     |                                                                         0|117.24s     |
|✅ MARKDOWN     |[markdown-table-formatter](https://megalinter.github.io/beta/descriptors/markdown_markdown_table_formatter)|  196|    0|                                                                         0|4.37s       |
|✅ OPENAPI      |[spectral](https://megalinter.github.io/beta/descriptors/openapi_spectral)                                 |    1|     |                                                                         0|1.51s       |
|:warning: PYTHON|[bandit](https://megalinter.github.io/beta/descriptors/python_bandit)                                      |  142|     |[**64**](https://github.com/megalinter/megalinter/actions/runs/1488844712)|2.42s       |
|✅ PYTHON       |[black](https://megalinter.github.io/beta/descriptors/python_black)                                        |  142|    0|                                                                         0|3.07s       |
|✅ PYTHON       |[flake8](https://megalinter.github.io/beta/descriptors/python_flake8)                                      |  142|     |                                                                         0|2.35s       |
|✅ PYTHON       |[isort](https://megalinter.github.io/beta/descriptors/python_isort)                                        |  142|    0|                                                                         0|0.5s        |
|✅ PYTHON       |[mypy](https://megalinter.github.io/beta/descriptors/python_mypy)                                          |  142|     |                                                                         0|63.0s       |
|✅ PYTHON       |[pylint](https://megalinter.github.io/beta/descriptors/python_pylint)                                      |  142|     |                                                                         0|15.91s      |
|✅ SPELL        |[cspell](https://megalinter.github.io/beta/descriptors/spell_cspell)                                       |  466|     |                                                                         0|38.83s      |
|✅ SPELL        |[misspell](https://megalinter.github.io/beta/descriptors/spell_misspell)                                   |  312|    0|                                                                         0|0.42s       |
|✅ XML          |[xmllint](https://megalinter.github.io/beta/descriptors/xml_xmllint)                                       |    2|     |                                                                         0|0.21s       |
|✅ YAML         |[prettier](https://megalinter.github.io/beta/descriptors/yaml_prettier)                                    |   77|    0|                                                                         0|3.86s       |
|✅ YAML         |[v8r](https://megalinter.github.io/beta/descriptors/yaml_v8r)                                              |   20|     |                                                                         0|28.52s      |
|✅ YAML         |[yamllint](https://megalinter.github.io/beta/descriptors/yaml_yamllint)                                    |   78|     |                                                                         0|1.45s       |


See errors details in [**artifact MegaLinter reports** on GitHub Action page](https://github.com/megalinter/megalinter/actions/runs/1488844712)

**Comment by Kurt-von-Laven** [comment]
I am curious whether upgrading rstfmt and rstcheck would address the build errors at this point, so I clicked "Update branch."

**Comment by Kurt-von-Laven** [comment]
Well, that changed the error message:

```
Error: Command failed: mega-linter-runner --path ./.. --release test-Kurt-von-Laven-updatePython -e "ENABLE=YAML" --nodockerpull
Unable to find image '***/***:test-Kurt-von-Laven-updatePython' locally
docker: Error response from daemon: manifest for ***/***:test-Kurt-von-Laven-updatePython not found: manifest unknown: manifest unknown.
See 'docker run --help'.
```

Is this possibly a build pipeline issue?

**Comment by nvuillam** [comment]
The issue is probably from the docker build step before :)

**Comment by nvuillam** [comment]
@Kurt-von-Laven FYI -> https://github.com/sgerrand/alpine-pkg-glibc/issues/169 :(

**Comment by Kurt-von-Laven** [comment]
Oof; I see. I wish it were released via some stable package manager, but personally I don’t know that I want MegaLinter to be in the business of packaging its dependencies at this stage of the project since there are so many wonderful linters that we could add instead for less effort. Maybe we can cache the package since GitHub Actions recently expanded its cache limit to 10 GB? Or maybe add retry logic and/or incremental downloading in case the connection fails? What would happen if we eliminated the dependency altogether?

**Comment by nvuillam** [comment]
I think it's used only for dart linter...
Do you know a way to cache a package ? Or even to not verify it with GPG key ? If yes it could be great ^^

**Comment by nvuillam** [comment]
By the way, with python 3.10, only test failures are rstfcheck and rstfmt.
Do people still use rst format ? :D 
Maybe we could get rid of those linters in v6 :)

 
FAILED ***/tests/test_***/linters/rst_rstcheck_test.py::rst_rstcheck_test::test_get_linter_help
FAILED ***/tests/test_***/linters/rst_rstcheck_test.py::rst_rstcheck_test::test_get_linter_version
FAILED ***/tests/test_***/linters/rst_rstcheck_test.py::rst_rstcheck_test::test_success
FAILED ***/tests/test_***/linters/rst_rstfmt_test.py::rst_rstfmt_test::test_get_linter_help
FAILED ***/tests/test_***/linters/rst_rstfmt_test.py::rst_rstfmt_test::test_get_linter_version
FAILED ***/tests/test_***/linters/rst_rstfmt_test.py::rst_rstfmt_test::test_success

**Comment by Kurt-von-Laven** [comment]
I might be missing something, but I believe we can simply use [GitHub's cache action](https://github.com/actions/cache/#implementation-examples). Is `apk add --no-cache glibc-${GLIBC_VERSION}.apk` the step that is verifying the GPG key? I'm struggling to find comprehensive documentation on that command, but I wonder what `--allow-untrusted` does after seeing https://wiki.alpinelinux.org/wiki/Package_management#Add_a_Package?

TL;DR I wholeheartedly agree about removing rstcheck and rstfmt for v6.

[StackOverflow has 1,155 questions tagged reStructuredText](https://stackoverflow.com/questions/tagged/restructuredtext), so it sounds like it's a documentation format that is essentially only used by a small handful of Python developers. There were only 8 hits for [rstcheck](https://stackoverflow.com/search?q=rstcheck), so it looks like most folks using reStructuredText don't use rstcheck, and like you say the project is no longer seeing releases. rstfmt had a release 3 months ago, but [their README](https://github.com/dzhu/rstfmt#rstfmt-a-formatter-for-restructuredtext) opens with: "Highly experimental and unstable. Do not depend on this yet." My intuition overall is that MegaLinter users as a whole will probably prefer having a lot of tools that play nice together out of the box to having even more tools but also more headaches. On top of that, we could easily end up getting completely bogged down by maintenance if we aren't careful. I'm sure we will hear from our users if reStructuredText is very important to them, .

**Comment by nvuillam** [comment]
Agreed, let's kill them in next version :D
